src/geometry/FlagSceneNode.cxx

```diff
@@ -89,39 +90,37 @@ const float WaveGeometry::RippleSpeed1 = (float)(2.4 * M_PI);
 const float WaveGeometry::RippleSpeed2 = (float)(1.724 * M_PI);
 
 
-inline void WaveGeometry::executeNoList() const
+inline void WaveGeometry::execute(bool shadow) const
 {
-    glDisableClientState(GL_NORMAL_ARRAY);
-    glVertexPointer(3, GL_FLOAT, 0, verts);
-    glTexCoordPointer(2, GL_FLOAT, 0, txcds);
-    glDrawArrays(GL_TRIANGLE_STRIP, 0, (flagChunks + 1) * 2);
-    glEnableClientState(GL_NORMAL_ARRAY);
-    return;
-}
-
-inline void WaveGeometry::execute() const
-{
-    if (flagLists)
-        glCallList(glList);
+    if (shadow)
+        vboVT.enableVertexOnly();
     else
-        executeNoList();
+        vboVT.enableArrays();
+    glDrawArrays(
+        GL_TRIANGLE_STRIP,
+        vboIndex,
+        (flagChunks + 1) * 2);
     return;
 }
 
 
-WaveGeometry::WaveGeometry() : refCount(0)
+WaveGeometry::WaveGeometry() : refCount(0), vboIndex(0)
```

Comment:
this is an error. `vboIndex` must be < 0 if not allocated.

And now I'm confused. The only places where this is actually used are here (note the buffer is never freed), MeshDrawMgr, and MeshFragSceneNode. Yet RenderNode is a VBOclient; one might expect that all RenderNodes should be allocating (and releasing) GL memory?

Why just these 3 places.

(by the way, we are stepping on each other at this point; I'm modifying this file as well).

Reply:
These three places now is because they were using something that is incompatible with the changes: 
GL Client buffers
a lot more will come next, any glBegin/glEnd should use this classes, gl list should go ...

Reply:
all RenderNodes will allocate GL memory, but not now. I don't want to put in this pull request all bzflag graphic code